app/mailers/user_mailer.rb

```diff
@@ -422,6 +400,17 @@ def open_project_headers(hash)
 # Unfortunately, this results in changes on the interceptor classes during development mode
 # not being reflected until a server restart.
 
+class PlainTextInterceptor
+  def self.delivering_email(message)
+    if Setting.plain_text_mail?
+      # HACK: this is a workaround to ensure @html_part ivar is set correctly
+      # on the Mail::Message.
+      message.html_part = message.html_part
+      message.html_part = nil
```

Comment:
Care to explain?
This looks like you always set the variable to `nil`.

On the one hand you already wrote this is a hack, but I still can't believe that this is what you intended.